### PR TITLE
feat(dmc): add DMC maintenance and engineering test modes (including DMC switching)

### LINKED ISSUE #3771: ECAM EIS Text bug
**Version**
Custom FBW (72c965f)

**Describe the bug**
When Testing EIS using the MCDU, "Maintenance Mode" is printed small;
In test mode, the text color is not enough visible

**To Reproduce**
Test EIS 1,2,3 and on switching panel, Potentiometer n°3 should be set to either 1 or 3

**Expected Behaviour**
The text should be bigger in Maintenance Mode, and text should be darker in Test Mode

ecam
ecam2

### PR DESCRIPTION
**Pull request by flogross89**
Fixes #3771

## Summary of Changes
- Re-introduced maintenance and engineering test modes in new typescript-based architecture
- Connect DMC switching knob to behavior of both modes (i.e. if DMC3 maint. mode is activated and DMC switching is set to F/O, the F/O PFD and ND display the "MAINTENANCE MODE" message

## Screenshots (if necessary)

No before screenshots (didn't work before with new instrument architecture)
test-mode-dmc1
test-mode-dmc-knob

## References

https://github.com/flybywiresim/a32nx/issues/3771
FCOM: DSC-31-05-30, DSC-31-05-40

## Additional context

Discord username (if different from GitHub):
floridude#7158

## Testing instructions

1) Power on airplane with BAT ON and EXT PWR ON. After initialization, select DMC display test modes in CFDS (SYSTEM TEST -> INSTRUMENTS -> ...). Cycle through all DMCs, and all DMC switching knob states. Only DUs receiving data from the respective DMCs should display the maintenance mode message or engineering test mode image. 

2) Perform normal flight from cold and dark until CLB, confirm all displays work normally without regression.

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

File: fbw-a32nx/src/systems/instruments/src/MsfsAvionicsCommon/displayUnit.tsx [hunk 1]
```diff
@@ -53,6 +63,21 @@ export class DisplayUnit extends DisplayComponent<DisplayUnitProps> {
             if (!document.documentElement.classList.contains('animationsEnabled')) {
                 document.documentElement.classList.add('animationsEnabled');
             }
+
+            const dmcKnob = SimVar.GetSimVarValue('L:A32NX_EIS_DMC_SWITCHING_KNOB', 'Enum');
```
Comment (flogross89) [inline fbw-a32nx/src/systems/instruments/src/MsfsAvionicsCommon/displayUnit.tsx:67]: Couldn't subscribe to changes on A32NX_EIS_DMC_SWITCHING_KNOB, so I had to resort to this solution.

**Comment by Benjozork** [comment]
Looks like the font used is wrong, should be ECAMFontRegular

**Comment by flogross89** [comment]
> Looks like the font used is wrong, should be ECAMFontRegular

You're right, thanks for the hint. It's hard to decipher using my reference [1], but it seems that way. Now updated: 
test-ecam-font

[1] https://www.youtube.com/watch?v=vNDJAFTdCKo

File: fbw-a32nx/src/systems/instruments/src/MsfsAvionicsCommon/common.scss [hunk 2]
```diff
@@ -17,7 +17,36 @@
...
+.EngineeringTestMode {
+  position: absolute;
+  left: 0%;
+  top: 0%;
+  width: 100%;
+  height: 100%;
+  border: none;
+  z-index: 999;
+  visibility: visible !important;
+  background-image: url('./thalesTest.svg');
```
Comment (Benjozork) [inline fbw-a32nx/src/systems/instruments/src/MsfsAvionicsCommon/common.scss:43]: This image should be placed in the  `/Images/fbw-a32nx/` VFS folder. To achieve this, you can place it appropriately in `fbw-a32nx/src/base/flybywire-aircraft-a320-neo/html_ui`, and use the first path I mentioned in this style.

File: fbw-a32nx/src/systems/instruments/src/Common/common.scss [hunk 3]
```diff
@@ -19,3 +19,32 @@
...
+
+.EngineeringTestMode {
+  position: absolute;
+  left: 0%;
+  top: 0%;
+  width: 100%;
+  height: 100%;
+  border: none;
+  z-index: 999;
+  background-image: url('./thalesTest.svg');
```
Comment (Benjozork) [inline fbw-a32nx/src/systems/instruments/src/Common/common.scss:41]: Same here.

**Review by Benjozork — CHANGES_REQUESTED** [review]
LGTM, just couple of changes needed to follow desired convention

File: fbw-a32nx/src/systems/instruments/src/MsfsAvionicsCommon/common.scss (hunk 2 above)
Comment (flogross89) [reply fbw-a32nx/src/systems/instruments/src/MsfsAvionicsCommon/common.scss:43]: Thanks for the pointer! I'm fairly new to this project. Fixed in latest commit.

**Review by Benjozork — APPROVED** [review]
Good for testing!

**Comment by mico975** [comment]
Quality Assurance Tester Report

Discord          : mico#3145
Object of testing: #7901
Tier of Testing  : 1
Date             : 29/03/2023

Testing Process: 
Power on airplane with BAT ON and EXT PWR ON. After initialization, select DMC display test modes in CFDS (SYSTEM TEST -> INSTRUMENTS -> ...). Cycle through all DMCs, and all DMC switching knob states. Only DUs receiving data from the respective DMCs should display the maintenance mode message or engineering test mode image. - **OK**

Perform normal flight from cold and dark until CLB, confirm all displays work normally without regression. - **OK**

Negatives:       
N/A

Testing Results: 
Passed

Conclusions: 
LGTM

Media:       
image

**Pull request merged**